book/chrome.md

```diff
@@ -119,267 +130,178 @@ Here\'s how I want inline layout to work, at a high level:
+    def flush(self):
+        child = self.children[-1]
+        child.x = self.x
+        child.y = self.cy
+        child.layout()
+        self.cy += child.h
+        self.children.append(LineLayout(self.node, self))
 ```
 
-Recall that for a `TextLayout` we compute the width and height in the
-constructor. Now, for a line, we need to just lay out the words in the
-line, one by one:
+Meanwhile the new `layout` method for `LineLayout`s is nearly the same
+as the old `flush` method, except:
 
-``` {.python}
-class LineLayout:
-    def layout(self, y):
-        self.y = y
-        self.x = self.parent.x
-        self.h = 0
-
-        x = self.x
-        leading = 2
-        y += leading / 2
-        for child in self.children:
-            child.layout(x, y)
-            x += child.w + child.space
-            self.h = max(self.h, child.h + leading)
-        self.w = x - self.x
-```
+1. It need to compute a `w` field, via `parent.w`
+2. It must loop over its children, instead of a `line` field.
+2. It needs to compute `x` and `y` fields on each child instead of
+   adding them to a display list
+3. Instead of updating the `cy` field, it must compute an `h` field.
 
-::: {.further}
-Note the height computation. It will be totally wrong if you mix fonts
-of different sizes in one line. You should instead first compute the
-largest ascenders and descenders, use that to compute a baseline, then
-place all the boxes, and finally compute the line height. Leading
-would be computed per-word and would factor into the placement of the
-baseline. I\'m not doing any of that here because we don\'t have any
-elements of different font sizes anyway.[^8]
-:::
-
-Now that we have words and lines laying themselves out, we need only
-modify `InlineLayout`. This involves the most surgery, but the end
-result is much simpler now that we\'ve got proper line and text
-layout.
-
-First, we can delete the `bold`, `italic`, `terminal_space`, and `dl`
-fields; pass `node` to the `InlineLayout` constructor; and rename the
-`layout` method to `recurse`:
+One annoyance is that, since `InlineLayout` always adds a new line in
+`flush`, the last flush call will add a new, unwanted line to the end.
+Let's just throw it away:
 
 ``` {.python}
 class InlineLayout:
-    def __init__(self, parent, node):
+    def layout(self):
         # ...
-        self.node = node
-
-    def recurse(self, node):
-        if isinstance(node, ElementNode):
-            for child in node.children:
-                self.recurse(child)
-        else:
-            self.text(node)
+        self.children.pop()
 ```
 
-This makes room for a new `layout` function, which calls `recurse` to
-create children and then lays them out:
+Now that words and lines lay themselves out, a lot of stuff disappears
+from`InlineLayout`. The `style`, `size`, `weight` fields are no longer
+needed. Neither is `line` or `cx`. And the display list is no longer
+needed: instead of addings words to a display list, they can print
```

Comment:
Print themselves?

Reply:
Yeah, this needs rewording. Something like,

> And the display list is no longer needed: instead of an `InlineLayout` drawing all the words in a paragraph, each word now corresponds to a `TextLayout` object that can draw it.

Reply:
LGTM